staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -782,6 +782,11 @@ func (c *Cacher) dispatchEvents() {
 }
 
 func (c *Cacher) dispatchEvent(event *watchCacheEvent) {
+	// FIXME: Introduce caching, but only at this level for now.
+	if event.Type != watch.Bookmark {
+		event.Object = newCachingObject(event.Object)
+	}
+
```

Comment:
If you had access to the serializer here in this function, would it still be necessary to inject the wrapper? Or could you just directly place the serialized bytes into the watchEvent.Object as e.g. a RawExtension object?

Reply:
I think it wouldn't help that much:
1. I don't want to serialize in this thread because there is already significant contention here (calls to this function are serialized)
2. This function is called once per event per apiserver - there can be multiple watchers, watching in different GV, so we would need to know all versions to which serialize

So I actually think this would have more drawbacks than benefits.